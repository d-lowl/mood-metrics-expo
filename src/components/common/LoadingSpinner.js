import React, { Component } from 'react';
import { View } from 'react-native';
import { Spinner } from 'native-base';


class LoadingSpinner extends Component {
  render() {
    return (
      <View>
        {this.props.children}
        <Spinner />
      </View>
    );
  }
}

export default LoadingSpinner;
