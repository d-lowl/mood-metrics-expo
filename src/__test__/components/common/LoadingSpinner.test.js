import 'react-native';
import React from 'react';
import LoadingSpinner from '../../../components/common/LoadingSpinner';
import { Text } from 'native-base';

import renderer from 'react-test-renderer';

test('Loading spinner renders correctly', () => {
  const tree = renderer.create(
    <LoadingSpinner />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

test('Loading spinner with text renders correctly', () => {
  const tree = renderer.create(
    <LoadingSpinner>
      <Text>Some text</Text>
    </LoadingSpinner>
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
