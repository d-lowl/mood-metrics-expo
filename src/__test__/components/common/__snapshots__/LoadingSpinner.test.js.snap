// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading spinner renders correctly 1`] = `
<View>
  <ActivityIndicator
    animating={true}
    color="#45D56E"
    hidesWhenStopped={true}
    size="large"
    style={
      Object {
        "height": 80,
      }
    }
    virtual={undefined}
  />
</View>
`;

exports[`Loading spinner with text renders correctly 1`] = `
<View>
  <Text
    accessible={true}
    allowFontScaling={true}
    disabled={false}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#000",
        "fontFamily": "System",
        "fontSize": 16,
      }
    }
    uppercase={false}
    virtual={undefined}
  >
    Some text
  </Text>
  <ActivityIndicator
    animating={true}
    color="#45D56E"
    hidesWhenStopped={true}
    size="large"
    style={
      Object {
        "height": 80,
      }
    }
    virtual={undefined}
  />
</View>
`;
